### PR TITLE
book theme layout

### PR DESCRIPTION
**Pull request by choldgraf**
I spent a little while hacking around our HTML template to make it look closer to Jupyter Book. I think that it's getting closer! It still needs some cleaning up, but after that I'm happy to either keep iterating on this PR if @rossbar has comments, or just merge it and we can keep incrementally iterating from there.

### CODE REVIEW

**Comment by rossbar** [comment]
Looks like a lot of great work on the look/layout!

Having all of the source files from the features section was useful as a test case for the `toc` stuff discussed in #2, but that's more of a "guts" concern than anything that affects the layout. I can always keep a branch on the side for working on that and not pollute master with it.

+1 for merge now and incrementally iterate

**Comment by choldgraf** [comment]
@rossbar yeah I agree, I just removed a bunch because rebuilding the html over and over was taking too long. We should just choose the minimal collection of files that demonstrates the features we want to make sure work in the build system

File: docs/content/features/beta.md [hunk 1]
```diff
@@ -1,6 +0,0 @@
-# Beta features
```
Comment (rossbar) [inline docs/content/features/beta.md]: Keep this file for testing `toc` nesting

File: docs/content/features/features.md [hunk 2]
```diff
@@ -1,15 +0,0 @@
----
```
Comment (rossbar) [inline docs/content/features/features.md]: Keep this file for testing `toc` nesting

File: docs/content/features/index.rst [hunk 3]
```diff
@@ -1,8 +0,0 @@
-==================
```
Comment (rossbar) [inline docs/content/features/index.rst]: Keep this file for testing `toc` nesting

File: docs/content/features/interact.ipynb [hunk 4]
```diff
@@ -1,119 +0,0 @@
-{
```
Comment (rossbar) [inline docs/content/features/interact.ipynb]: Keep this file for testing `toc` nesting

**Review by rossbar — COMMENTED** [review]
I vote for keeping a subset of the features files so that we can experiment with approaches for specifying the `toc` without increasing the build time too much. I've earmarked 4 files that I think should do the trick.

As I mentioned before however, if you want to just merge into master as is and focus on the layout components that's totally fine - I can keep a branch in my fork for `toc` testing then cherry-pick any relevant changes if something comes out of it.

**Comment by choldgraf** [comment]
that sounds like a good compromise to me, I'll add those in before merging

**Comment by choldgraf** [comment]
OK the latest push re-includes the files that @rossbar mentioned...do those look OK?

It also adds another nifty thing I figured out, which is how to include the key/vals from a `config.yml` file into the HTML context so that the templates can use it. This would let us keep the same configuration pattern for the site variables and avoid users needing to write Python code.

**Review by rossbar — APPROVED** [review]
LGTM - Very cool re: the yaml->html ctx, seems like a pattern that will be quite useful more generally

**Comment by choldgraf** [comment]
OK I'm gonna merge this and we can keep iterating

**Pull request merged**